self-serve concurrency in HCPTF (#3152)
hashicorp/web-unified-docs: content/terraform-docs-common/docs/cloud-docs/overview/index.mdx

Reviewer: ```suggestion
HCP Terraform organizes infrastructure into projects that contain workspaces and Stacks. You can use projects to organize your workspaces and Stacks into groups. Organizations with HCP Terraform [Essentials](https://www.hashicorp.com/products/terraform/pricing) edition can assign teams permissions for specific projects.
```

The workspaces/stacks info earlier means this can be simplified.

Not sure exactly where the Essentials bit came from, but maybe it'd be more accurate to say Essentials or higher, or a paid plan?

---
page_title: HCP Terraform subscription plans
description: >-
  Learn about HCP Terraform subscription plans including free, Essentials, 
  Standard, and Premium. Compare features, prices, technical
  limits, and learn about managed resources.
tfc_only: true
# START AUTO GENERATED METADATA, DO NOT EDIT
created_at: 2025-05-27T14:28:51-04:00
last_modified: 2026-08-19T22:25:44.000Z
# END AUTO GENERATED METADATA
---

[cli]: /terraform/cli
[speculative plans]: /terraform/cloud-docs/workspaces/run/remote-operations#speculative-plans
[remote_state]: /terraform/language/state/remote-state-data
[outputs]: /terraform/language/values/outputs
[modules]: /terraform/language/modules/develop
[terraform enterprise]: /terraform/enterprise

# HCP Terraform subscription plans

> HCP Terraform is a platform that performs Terraform runs to provision infrastructure, either on demand or in response to various events. Unlike a general-purpose continuous integration (CI) system, it is deeply integrated with Terraform's workflows and data.

Reviewer: ```suggestion
HCP Terraform is a platform that uses Terraform to provision and manage infrastructure, either on demand or in response to various events. Unlike a general-purpose continuous integration (CI) system, it is deeply integrated with Terraform's workflows and data.
```

> **Hands On:** Try our [What is HCP Terraform - Intro and Sign Up](/terraform/tutorials/cloud-get-started/cloud-sign-up) tutorial.

## Subscription plans

HCP Terraform offers free, Essentials, Standard, and Premium subscription plans. Each plan provides different features and resource limits to match your team's needs.

HCP Terraform manages plans and billing at the [organization level](/terraform/cloud-docs/users-teams-organizations/organizations). Each HCP Terraform user can belong to multiple organizations, which might subscribe to different billing plans. The set of features available depends on which organization you work in.

<Note>

HCP Europe organizations support specific contract billing plans.

@include 'eu/billing.mdx'

</Note>

### Free plan

> Small teams can use most of HCP Terraform's features for free, including remote Terraform execution, VCS integration, the private module registry, single-sign-on, policy enforcement, and run tasks.

Reviewer: It's okay to leave as-is, but it occurs to me that it's not really "small teams" so much as "teams with a relatively small amount of managed infrastructure".  I can't think of a good, non-wordy way to say that.

> Small teams can use most of HCP Terraform's features for free, including remote Terraform execution, VCS integration, the private module registry, single-sign-on, policy enforcement, and run tasks.

Author: ```suggestion
Teams getting started with infrastructure-as-code can use most HCP Terraform's features for free, including remote Terraform execution, VCS integration, the private module registry, single-sign-on, policy enforcement, and run tasks.
```

Free organizations are limited to 500 managed resources. This plan supports individuals, small teams, and organizations evaluating HCP Terraform.

> ### Essentials plan

Reviewer: I like this refactor.  The headings and table make it so much easier to figure out what the plans are and how they're different.

## Plan comparison

The following table compares features across HCP Terraform plans. Refer to [Terraform pricing](https://www.hashicorp.com/products/terraform/pricing) for details and pricing:

| Feature | Free | Essentials | Standard | Premium |
|---------|------|------------|----------|---------|
| Managed resources | Up to 500 | Unlimited | Unlimited | Unlimited |
| Remote Terraform execution | ✓ | ✓ | ✓ | ✓ |
| VCS integration | ✓ | ✓ | ✓ | ✓ |
| Private module registry | ✓ | ✓ | ✓ | ✓ |
| Team management | ✓ | ✓ | ✓ | ✓ |
| Run tasks | ✓ | ✓ | ✓ | ✓ |
| Concurrent runs | 1 remote, 1 agent | 3 remote, 3 agent | 10 remote, 10 agent | 30 remote, 30 agent |
| Policy enforcement | - | - | ✓ | ✓ |
| Cost estimation | - | - | ✓ | ✓ |
| Audit logging | - | - | - | ✓ |
| Custom RBAC | - | - | - | ✓ |
| SSO (SAML) | - | - | ✓ | ✓ |
| Priority support | - | - | - | ✓ |

## Billing models

HCP Terraform offers two billing models to match different organizational needs.

<Note>

HCP Europe organizations support specific contract billing plans.

@include 'eu/billing.mdx'

</Note>

### Pay-as-you-go

> Pay-as-you-go billing charges based on your actual usage of managed resources. This model supports organizations with variable infrastructure needs or those wanting to start small and scale gradually.

Reviewer: ```suggestion
Pay-as-you-go billing charges monthly based on your actual usage of managed resources. This model supports organizations with variable infrastructure needs or those wanting to start small and scale gradually.
```

To activate pay-as-you-go billing, refer to [Activate pay-as-you-go](/terraform/cloud-docs/overview/activate-paygo).

### HashiCorp Flex

HashiCorp Flex provides committed usage pricing with volume discounts. This model supports organizations with predictable infrastructure needs who want to optimize costs through annual commitments.

Flex plans offer greater discounts at larger scale compared to pay-as-you-go billing. Contact your HashiCorp account team to discuss Flex options.

To activate Flex billing, refer to [Activate Flex](/terraform/cloud-docs/overview/activate-flex).

### Contract plans

Organizations with enterprise agreements can work with HashiCorp sales to create custom contract plans. Contract plans can include volume discounts, committed usage, and bundled features tailored to your organization's needs.

[Contact HashiCorp Sales](https://www.hashicorp.com/contact-sales) to discuss contract options.

## Pricing

HCP Terraform pricing varies by plan and billing model. The following sections explain how pricing works and provide example calculations.

### How pricing works

> HCP Terraform pricing is based on managed resources for Essentials, Standard, and Premium plans. HashiCorp charges for each managed resource on a per-hour basis, from the time a managed resource is provisioned until it is destroyed.

Reviewer: It just struck me funny that Essentials ends in an `s`, but none of the other plan names do.  Huh.

Each partial hour is billed as a full hour. The peak number of managed resources in a given hour determines the cost for that hour.

### Pricing by plan

> - Free plan: No charge for up to 500 managed resources.

Reviewer: ```suggestion
HCP Terraform pricing is based on your plan:

- Free plan: No charge for up to 500 managed resources.
```

- Essentials plan: Pay-as-you-go billing per managed resource per hour. Refer to the [pricing page](https://www.ibm.com/products/hashicorp/pricing) for current rates.
- Standard and Premium plans: Contact [HashiCorp Sales](https://www.hashicorp.com/contact-sales) for pricing information.

Refer to [Estimate HCP Terraform cost](/terraform/cloud-docs/overview/estimate-cost) for more information, including an example cost calculation.

### Billing cycles

HCP Terraform bills monthly based on your usage during the billing period. You can view your current usage and billing information in the [HashiCorp Cloud Platform portal](/hcp/docs/hcp/admin/billing#usage), which refreshes hourly.

> ## Managed resources

Reviewer: This is a really good place for this, much better than on the cost page where it was.  Nice!

### What is a managed resource

A **managed resource** is a resource in an HCP Terraform-managed state file where `mode = "managed"`. HCP Terraform counts a resource as part of this count starting from the first `terraform plan` or `terraform apply` operation on the resource.

HCP Terraform combines workspace and Stacks resources under management when calculating costs.

Examples of managed resources include:

- Resources provisioned by a Terraform provider (for example, an AWS VPC)
- Resources created by [count](/terraform/language/meta-arguments/count) and [for_each](/terraform/language/meta-arguments/for_each) meta-arguments
- Resources provisioned by [modules](/terraform/language/modules) and [no-code ready modules](/terraform/cloud-docs/workspaces/no-code-provisioning/module-design)

### What does not count as a managed resource

HCP Terraform does not include the following in the total managed resource count:

- Resources defined as a `null_resource`
- Resources defined as a `terraform_data` resource
- Data sources (resources where `mode = "data"`)
- Local values and variables

> ### View your managed resource count

Reviewer: This section also exists on the Estimate Cost page.  I think it makes more sense there.  Replace this section with a link?

Sign in to [HCP Terraform](https://app.terraform.io/) and choose **Usage** from the sidebar to navigate to a report of your organization's total number of managed resources. You can use the total number of managed resources to determine your number of billable managed resources in use at any given time.

The Usage report provides an at-a-glance view of general usage limits in the organization:

- A summary of projects, workspaces, Stacks, and applies
- General limits for managed resources, concurrency, and agents

> - Compliance feature set limits for run tasks and policies

Author: ```suggestion

You can open the **Usage report** page for an at-a-glance view of resource usage and other information. Refer to [Estimate costs](/terraform/cloud-docs/overview/estimate-cost) for more information.
```

## Technical limits

Each HCP Terraform plan has specific technical limits for resources, concurrency, API usage, teams, and governance features.

### Resource limits

> 

Reviewer: Noting that some of these are redundant with the plan comparison table, but there is more detail in these.  Also noting that it's convenient to not have to scroll back up for the repeated data.  So I think it's all good.

No change needed.

| SSO (SAML) | Not available | Not available | ✓ | ✓ |
| Custom RBAC | Not available | Not available | Not available | ✓ |

### Policy and governance limits

| Limit | Free | Essentials | Standard | Premium |
|-------|------|------------|----------|---------|
| Policy sets | Not available | Not available | Unlimited | Unlimited |
| Policies per set | Not available | Not available | Unlimited | Unlimited |
| Cost estimation | Not available | Not available | ✓ | ✓ |
| Audit logging | Not available | Not available | Not available | ✓ |

## Terraform workflow

HCP Terraform runs the [Terraform CLI][cli] to provision infrastructure. In its default state, the Terraform CLI uses a local workflow, performing operations on the workstation where it is invoked and storing state in a local directory.

HCP Terraform lets you organize and manage infrastructure using workspaces or Stacks.

- Workspaces manage a self-contained infrastructure of one Terraform root module.
- Stacks manage multiple infrastructure modules and repeat that infrastructure at scale.

To learn whether a workspace or a Stack fits your use case, refer to [Choose workspaces or Stacks](/terraform/cloud-docs/stack-workspace).

> Because teams must share responsibilities and awareness to avoid single points of failure, working with Terraform in a team requires a remote workflow.

Reviewer: ```suggestion
Working with Terraform in a team requires a remote workflow.
```

Might be too terse, and I'm okay with not taking the suggestion.  But I can think of a lot of other reasons than what was already there that you need a remote workflow (e.g., "unless y'all want to run Terraform only from Jimbob's laptop...")

VCS integration is optional. If you use an unsupported VCS or want to preserve an existing validation and deployment pipeline, you can use the API or Terraform CLI to upload new configuration versions. You'll still get the benefits of remote execution and HCP Terraform's other features.

Refer to the following topics for more information:

- [VCS-driven workspace runs](/terraform/cloud-docs/workspaces/run/ui)
- [Stack runs](/terraform/cloud-docs/stacks/runs).
- [Supported VCS Providers](/terraform/cloud-docs/vcs#supported-vcs-providers)

### Command line integration

Remote execution benefits teams, while local execution benefits individual developers. For example, most Terraform users run `terraform plan` to interactively check their work while editing configurations.

HCP Terraform allows you to run remote plans from your local command line.

Remote CLI-driven runs use the current working directory's Terraform configuration and the remote workspace's variables, so you don't need to obtain production cloud credentials to preview a configuration change.

The HCP Terraform CLI integration also supports state manipulation commands such as `terraform import` or `terraform taint`.

> If you are working with a Stack, you can use the `terraform stacks` commands to manage your Stack and its configuration, deployments, and more from the command line. Refer to [the `terraform stacks` commands](/terraform/cli/stacks) for more information.

Reviewer: ```suggestion
If you are working with a Stack, you can use the `terraform stacks` commands to manage your Stack and its configuration, deployments, and more from the command line. Refer to [`terraform stacks` reference](/terraform/cli/commands/stacks) for more information.
```

### Private Registry

Teams can benefit by codifying commonly used infrastructure patterns into reusable [modules][].

> Terraform can fetch providers and modules from many sources. HCP Terraform provides a private registry to find providers and modules. Users throughout your organization can browse a directory of internal providers and modules, and can specify flexible version constraints for the modules they use in their configurations. Versioning lets downstream teams use private modules with confidence, and frees upstream teams to iterate faster.

Reviewer: ```suggestion
Terraform can fetch providers and modules from many sources. HCP Terraform provides a private registry to find providers and modules. Users throughout your organization can browse a directory of internal providers and modules, and can specify flexible version constraints for the modules they use in their configurations. Versioning lets module consumers use private modules with confidence, and frees module creators to iterate faster.
```
